src/snowflake/cli/api/project/project_paths.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+from pathlib import Path
+from shutil import rmtree
+
+
+@dataclass
+class ProjectPaths:
+    project_root: Path
+
+    @property
+    def deploy_root(self) -> Path:
+        return self.project_root / "output"
```

Comment:
native apps uses output/deploy for this. We have other ephemeral directories under output/ as well, e.g. output/bundle as the bundle root. We also make all of those configurable.